lib/cn/csched_sp3.c

```diff
@@ -1151,7 +1151,7 @@ sp3_process_dirtylist(struct sp3 *sp)
                  * it until the next call to sp3_process_dirtylist().
                  */
                 mutex_lock(&sp->sp_dlist_lock);
-                busy = list_empty(&tn->tn_dnode_linkv[i]);
+                busy = !list_empty(&tn->tn_dnode_linkv[i]);
```

Comment:
This was a nasty bug, it was leaving empty nodes in the tree, and worse, it potentially could remove nodes that were still on the alternate dirty list.